core/trino-main/src/main/java/io/trino/execution/ColumnInfo.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import java.util.Optional;
+
+public record ColumnInfo(Optional<String> catalog, Optional<String> schema, Optional<String> table, String name, String label)
+{
+    public ColumnInfo(String name)
+    {
+        this(Optional.empty(), Optional.empty(), Optional.empty(), name, name);
+    }
+
+    public ColumnInfo(String name, String label)
+    {
+        this(Optional.empty(), Optional.empty(), Optional.empty(), name, label);
+    }
+
+    public ColumnInfo(String catalog, String schema, String table, String name)
+    {
+        this(Optional.of(catalog), Optional.of(schema), Optional.of(table), name, name);
+    }
+
+    public ColumnInfo(String catalog, String schema, String table, String name, String label)
+    {
+        this(Optional.of(catalog), Optional.of(schema), Optional.of(table), name, label);
+    }
```

Comment:
We avoid multiple overloaded constructors unless there is a good reason to have them.